### PR TITLE
aws.ec2.monitoring detection for [MITRE IaaS Gap] - T1204 User Execution

### PR DESCRIPTION
**Pull request by jstan12**
### Background

A single detection that covers these three aspects of T1204

### CODE REVIEW

File: rules/aws_cloudtrail_rules/aws_ec2_monitoring.yml [hunk 1]
```diff
@@ -0,0 +1,342 @@
+AnalysisType: rule
+Description: Checks for occurrences of EC2 Instance Creation, EC2 Instance Start or EC2 Image Creation within CloudTrail
+DisplayName: AWS EC2 Monitoring
+Enabled: true
+Filename: aws_ec2_monitoring.py
+Reports:
+    MITRE ATT&CK:
+        - TA:0002:T1204
```
Comment (edyesed) [inline rules/aws_cloudtrail_rules/aws_ec2_monitoring.yml:8]:
```suggestion
        - TA0002:T1204
```

**Review by edyesed — COMMENTED** [review]
I think the `ec2_instance` is already covered in this detection https://github.com/panther-labs/panther-analysis/blob/master/rules/aws_cloudtrail_rules/aws_modify_cloud_compute_infrastructure.py#L40-L63 

I also suggest adopting the highlighted block as the filter set to reduce FPs

**Review by maxrichie5 — DISMISSED** [review]

**Review by edyesed — CHANGES_REQUESTED** [review]
This will generate a substantial number of FPs without the filter set mentioned here https://github.com/panther-labs/panther-analysis/pull/545#pullrequestreview-1160638216. There are also an overlapping set of eventNames with the detection linked in the comment

**Comment by nkulig** [comment]
Hey Ed, since the detection you linked is overlapping, does it make sense to modify the EC2_CRUD_ACTIONS object to simply list EC2 instance creation or image creation? Or are you saying that the detection you linked could also be used for T1204?

**Comment by edyesed** [comment]
> Hey Ed, since the detection you linked is overlapping, does it make sense to modify the EC2_CRUD_ACTIONS object to simply list EC2 instance creation or image creation? Or are you saying that the detection you linked could also be used for T1204?

I would remove the `ec2_instance` grouping of eventNames in this detection. It makes sense to me to have this detection in addition to the linked one as the `*Image`  eventNames in this detection are related to a separate and distinct Tactic.

**Comment by nkulig** [comment]
Hey Ed! Let me know if I understood your suggestions correctly and if not I would be happy to add the necessary changes.

**Review by edyesed — APPROVED** [review]
detection for EC2 Image eventNames, this detection makes an attempt to ignore Servic-to-Service calls that invoke the Image eventNames. Test cases include an alert-generating AssumedRole based event 👍

**Pull request merged**